### DISCUSSION TITLE
Does Bevy support letterboxing for 2D cameras?

### DISCUSSION THREAD

**Discussion by Newbytee** [bevyengine/bevy#3127] [Q&A] [answered]
In #400 it's mentioned that this just is one of many ways to deal with windows being resized when working with 2D, with letterboxing being mentioned as one alternative. Is letterboxing currently available in Bevy?

**Comment by cart** [comment, marked as answer]
Not yet, but we're building the new renderer (coming in 0.6) with this type of scenario in mind.

**Reply by nicolas-sabbatini** [reply]
Now with the release of 0.6 and the new render this is possible?

**Reply by cart** [reply]
Not quite. The foundations have been laid, but we still need to implement this. Expect this to happen soon (probably by 0.7).

**Reply by Newbytee** [reply]
What about now that 0.7 is out?

**Reply by cart** [reply]
Not quite, but once #4745 lands (which should be very soon) it will be reasonable to build this out in userspace without writing a bunch of custom renderer code. Just point a camera at a texture of the desired size, then use that texture on a Bevy UI node that scales using your desired layout strategy.

That being said, after that its worth exploring even higher level apis to make this just a couple of lines of config.

**Reply by pixelprizm** [reply]
Obligatorily asking the same question now that 0.8 is out :) - Is this still the best way to do letterboxing? https://gist.github.com/cart/87c0215aa650170f09164fcffc252139
